spec/spec_helper.rb

```diff
@@ -1,7 +1,7 @@
 # typed: strict
 # frozen_string_literal: true
 
-require "tapioca"
+require "tapioca/all"
```

Comment:
I don't understand what we're doing here.

Since `tapioca` loads `tapioca/all`, can't we keep `require 'tapioca'`? 

What's the point of this commit?

Reply:
We've been doing the exports from the gem incorrectly up to now, and this commit is fixing it.

All the compilers and generators, etc are internal types of Tapioca and should not be loaded for clients that do a simple `gem "tapioca"` in their `Gemfile`. Those clients will end up requiring Tapioca and will get all kinds of compilers/generators loaded.

This didn't use to be a problem until now, since Tapioca was not modifying global state (much). Now that we are patching `T::Types::Simple` and `T::Types::Simple`, we really should not allow clients to load those classes unless they really need to.

The name `tapioca/all` for that optional require might not be the best. Maybe it should be `tapioca/internal` or something, I am open to ideas.

Basically:
- `gem 'tapioca'` loads `tapioca` which only loads the classes needed to write DSL generators.
- Running `tapioca` loads `cli.rb` which loads `tapioca/all` which loads everything that is needed to do a full Tapioca run.
- Similarly, spec helper loads `tapioca/all` so that test can find all the constants they expect.

Reply:
Makes sense, thanks for the explanation.

Yeah, if we want to discourage people to load this it should not be called `all`. `internal` is scary, I like it 👍

Reply:
I moved this into its own PR: https://github.com/Shopify/tapioca/pull/261